platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -358,69 +363,79 @@ - (MGLImage *)blurredAttributionBackground:(CIImage *)backgroundImage
+    switch (style) {
+        case MGLAttributionInfoStyleLong:
 #if TARGET_OS_IPHONE
-    largeLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+            logoImage =[UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
 #else
-    largeLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+            logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
 #endif
-    _defaultLogoHeight = largeLogoAttribution.logoImage.size.height;
-    
-    MGLSnapshotAttributionOptions *smallLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+            _defaultLogoHeight = logoImage.size.height;
+            break;
+            case MGLAttributionInfoStyleMedium:
 #if TARGET_OS_IPHONE
-    smallLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+            logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
 #else
-    smallLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
+            logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
 #endif
+            break;
+            
+        case MGLAttributionInfoStyleShort:
+            logoImage = nil;
+            break;
+    }
+    return logoImage;
+}
+
+- (CGSize)attributionSizeWithLogoStyle:(MGLAttributionInfoStyle)logoStyle sourceAttributionStyle:(MGLAttributionInfoStyle)attributionStyle
+{
+    MGLImage *logoImage = [self logoImageWithStyle:logoStyle];
     
-    MGLSnapshotAttributionOptions *noLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:YES];
+    CGSize attributionBackgroundSize = [self attributionTextSizeWithStyle:attributionStyle];
     
-    [options addObject:largeLogoAttribution];
-    [options addObject:smallLogoAttribution];
-    [options addObject:noLogoAttribution];
+    CGSize attributionSize = CGSizeZero;
     
-    for (MGLSnapshotAttributionOptions *attributionOptions in options) {
-        // -[Mapbox Logo]-[Attribution Background]-
-        CGFloat origin = attributionOptions.logoImage ? MGLLogoImagePosition.x : 0;
-        CGFloat width = origin + attributionOptions.logoImage.size.width + 10 + attributionOptions.attributionBackgroundSize.width + 10;
-        if (width <= snapshotSize.width) {
-            return attributionOptions;
-        }
+    if (logoImage) {
+        attributionSize.width = MGLLogoImagePosition.x + logoImage.size.width + 10;
```

Comment:
This line has no effect because the following line always overwrites `attributionSize.width`.

Reply:
And that is ok. In the following line:
`attributionSize.width = attributionSize.width + 10 + attributionBackgroundSize.width + 10;`

If has a logo adds its size and carry the value to sum the attribution text size.

Reply:
Gah, sorry, that makes sense. Didn’t see it for some reason.